sycl/tools/get_device_count_by_type.cpp

```diff
@@ -224,6 +228,49 @@ static bool queryCUDA(cl_device_type deviceType, cl_uint &deviceCount,
 #endif
 }
 
+static bool queryROCm(cl_device_type deviceType, cl_uint &deviceCount,
+                      std::string &msg) {
+  deviceCount = 0u;
+#ifdef USE_PI_ROCM
+  switch (deviceType) {
+  case CL_DEVICE_TYPE_DEFAULT: // Fall through.
+  case CL_DEVICE_TYPE_ALL:     // Fall through.
+  case CL_DEVICE_TYPE_GPU: {
+    int count = 0;
+    hipError_t err = hipGetDeviceCount(&count);
+    if (err != hipSuccess || count < 0) {
+      msg = "ERROR: ROCm error querying device count";
+      return false;
+    }
+    if (count < 1) {
+      msg = "ERROR: ROCm no device found";
+      return false;
+    }
+    deviceCount = static_cast<cl_uint>(count);
+#if defined(__HIP_PLATFORM_AMD__)
+    msg = "rocm-amd ";
+#elif defined(__HIP_PLATFORM_NVIDIA__)
+    msg = "rocm-nvidia ";
+#else
```

Comment:
Can we work with both AMD and NVIDIA at the same time?

Reply:
No, at the moment the ROCm plugin is only built for one or the other.

It is possible to have the ROCm plugin for AMD and the CUDA plugin for NVIDIA though, but that wouldn't affect `get_device_count_by_type`.